inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+                            double d = (float) document.getFirstValue("score");
+                            result.setScore(d);
+                        }
+
+                        fillResultWithMetadata(result, document, aTraits);
+
+                        if (response.getHighlighting().size() != 0) {
+                            List<ExternalSearchHighlight> highlights = new ArrayList<>();
+                            Map<String, Map<String, List<String>>> idHighlight =
+                                    response.getHighlighting();
+
+                            if (idHighlight.get(document.getFirstValue("id"))
+                                    .get(aTraits.getDefaultField()) != null) {
+                                for (String chaine : idHighlight.get(document.getFirstValue("id"))
+                                        .get(aTraits.getDefaultField())) {
+                                    highlights.add(new ExternalSearchHighlight(chaine));
+                                }
+                            }
+
+                            result.setHighlights(highlights);
+                        }
+                        results.add(result);
+                    }
+                    System.out.println("Found " + documents.getNumFound() + " documents");
+                }
+                catch (BaseHttpSolrClient.RemoteSolrException e) {
+                    throw new IOException(e.getMessage());
+                }
+            } catch (BaseHttpSolrClient.RemoteSolrException e) {
+                throw new IOException("Error : The search path does not exist");
+            }
+
+        } catch (BaseHttpSolrClient.RemoteSolrException e) {
+            throw new IOException("Error : The name of the collection does not exist");
+        }
+        catch (SolrServerException | HttpHostConnectException e)
+        {
+            throw new IOException(e.getMessage());
+        }
+
+        return results;
+    }
+    /**
+     * Convert data from SolrDocument to ExternalSearchResult
+     * @param result result has just the id complete before the methode
+     * @param document contain all the information about the document
+     * @param aTraits request parameters
+     */
+    private void fillResultWithMetadata(ExternalSearchResult result, SolrDocument document,
+                                        SolrSearchProviderTraits aTraits)
+            throws IOException, SolrException
+    {
+        if (isNotBlank((String) document.getFirstValue(DOC_NAME_KEY)))
+        {
+            result.setDocumentTitle((String) document.getFirstValue(DOC_NAME_KEY));
+        }
+        else if (isNotBlank((String) document.getFirstValue(DOC_TITLE_KEY)))
+        {
+            result.setDocumentTitle((String) document.getFirstValue(DOC_TITLE_KEY));
+        }
+        else
+            result.setDocumentTitle((String) document.getFirstValue(DOC_ID_KEY));
+
+        if (isNotBlank((String) document.getFirstValue(DOC_LANGUAGE_KEY)))
```

Comment:
The formatter will make quite a few changes in this method, e.g. adding curly braces in many places.